src/guard.rs

```diff
@@ -192,6 +192,38 @@ impl Guard {
             local.flush(self);
         }
     }
+
+    /// Unpins and then pins the current thread.
+    ///
+    /// Since `&mut self` is given, we can statically deduce that there is no [`Shared`] pointer
+    /// that is created before an invocation of `safepoint()` and then survives after the
+    /// invocation.
+    ///
```

Comment:
Should probably also mention that the current thread will be repinned only if this is the only active guard.